### DISCUSSION TITLE
Question: Is there an alternative to step 5 for the linux build from source?

### DISCUSSION THREAD

**Discussion by matdexir** [AppFlowy-IO/AppFlowy#682] [General]
Hi,

I was trying to build `linux' debug version` from source by following the instructions on https://appflowy.gitbook.io/ and I kept failing to run the binary until I saw that I missed `step 5`. 
As `non-vscode` user who is also considering contributing to the project, I was wondering if there's an alternative way to run the program, i.e without vs-code.
For instance: `make clean-debug` that can replicate the vscode Clean & Build command functionalities.
Thank you for your understanding!

### Error message

This is the error message, I get after running the binary directly after step 4.

![2022-07-22_13-40](https://user-images.githubusercontent.com/58453130/180370852-92b9bef6-012e-410a-89f8-d8b67a649d1c.png)

### Just in case
* OS version: Arch Linux x86_64 5.18.12-arch1-1
* Flutter version: Flutter 3.0.5 • channel stable
* Cargo version: cargo 1.62.1

**Comment by MikeWallaceDev** [comment]
Hi there, thanks for you interest in AF!  
We currently only focus on VS Code, however we're open to PRs if you want to code in something else.
You could see some of the command in step 3. You can find them all in the `"/appflowy/frontend/Makefile.toml"` file (and its includes)

For more information, come discuss on Discuss, and maybe we can improve the documentation.

**Reply by matdexir** [reply]
Thank you for this prompt reply!
I'll look into it ASAP.

**Reply by annieappflowy** [reply]
@matdexir you can join us on Discord https://discord.gg/9Q2xaN37tV 
@LucasXu0 is taking a look into your issue.

**Reply by matdexir** [reply]
@annieappflowy I just did that!

**Comment by LucasXu0** [comment]
Hi @matdexir. You could run the below commands after step 3.1 (`cargo make --profile development-linux-x86_64 flowy-sdk-dev`). 

Don't need to execute the `cargo make -p development-linux-x86_64 appflowy-linux-dev`.

```
cd frontend
// generate the languages.
sh ./scripts/generate_language_files.sh
// build the flutter project
cd app_flowy
flutter pub get
flutter packages pub get
flutter pub run build_runner build --delete-conflicting-outputs
flutter run
```

I could build the Appflowy project successfully without VSCode installed.

**Reply by matdexir** [reply]
Hi @LucasXu0
Thank you very much for your help! I was able to build it properly. 
A few corrections that I think may be valuable to the build process of the App.

You can only run `sh ./scripts/generate_language_files.sh` after you run `flutter pub get`

When it comes to dependencies, I would also advise to add `xdg-user-dirs`(on Arch, not sure about other distros) as one of them since it may not necessarily come installed on the end user's machine. 

Once again thanks for your help!

**Comment by MikeWallaceDev** [comment]
Hello everyone,
You don't _need_ VS Code, it's just that it's the IDE that we currently support. Building at the command line should already work flawlessly (I do it all the time  :) )

If you follow the build instructions you'll notice that the app is actually build at step 4. Step 5 is when you start editing the application in the IDE.

You should not have to run any commands other than the two cargo make commands.  If you found that it was required to run something other than those two commands, then something is wrong with our scripts.

I'm currently working on the development environment installation process for all 3 platforms, and will update the documentation accordingly.  So, if there's anything that you feel is missing, or can be better, just let me know :)
